ARCollectionViewMasonryLayout.m

```diff
@@ -316,14 +356,9 @@ - (UICollectionViewLayoutAttributes *)layoutAttributesForItemAtIndexPath:(NSInde
 
 - (NSArray *)layoutAttributesForElementsInRect:(CGRect)rect
 {
+    // Lays out all of the cells in the collection view
+    // extremely performance critical code.
     NSArray *attributes = self.attributesGrid.allItemAttributes;
-    if (self.headerAttributes) {
-        attributes = [attributes arrayByAddingObject:self.headerAttributes];
-    }
-    if (self.footerAttributes) {
-        attributes = [attributes arrayByAddingObject:self.footerAttributes];
-    }
-
```

Comment:
can be removed due to the invalidation work, don't need to pass in via the cell collection view api